db/database.go

```diff
@@ -15,6 +15,7 @@ type KeyValueReader interface {
 	//		return nil
 	//	})
 	Get(key []byte, cb func(value []byte) error) error
+	Iterable
```

Comment:
While I think `Iterable` is over abstraction, we should do either of these, depending on the relationship between Reader and Iterable:
- If Reader === Iterable: Remove `Iterable` and move its functions to `KeyValueReader`
- If Iterable is a subset of Reader: Iterable interface should contain `KeyValueReader` instead
- If Iterable and Reader are independent (current assumption): If we need to provide both, create another interface to provide both of them instead.